File: src/test/java/com/example/probation/controller/PagesControllerTest.java
package com.example.probation.controller;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.http.HttpStatus;
import org.springframework.test.context.jdbc.Sql;
import org.springframework.test.context.junit.jupiter.SpringExtension;
import org.springframework.test.web.servlet.MockMvc;

import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestBuilders.formLogin;
import static org.springframework.security.test.web.servlet.response.SecurityMockMvcResultMatchers.authenticated;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

@ExtendWith(SpringExtension.class)
@AutoConfigureMockMvc

@Sql({"/init.sql"})
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class PagesControllerTest {
    @Autowired
    private MockMvc mockMvc;

    @Test
    void login() throws Exception {
        mockMvc.perform(formLogin("http://localhost:8080/login").user("Admin").password("@Admin0000"))
                .andExpect(status().is(HttpStatus.FOUND.value()))
                .andExpect(authenticated());
    }
}

File: src/test/kotlin/com/example/probation/controller/PagesControllerTest.kt
package com.example.probation.controller

import org.junit.jupiter.api.Test
import org.springframework.http.HttpStatus
import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestBuilders
import org.springframework.security.test.web.servlet.response.SecurityMockMvcResultMatchers
import org.springframework.test.web.servlet.result.MockMvcResultMatchers
import org.springframework.web.context.WebApplicationContext

class PagesControllerTest(context: WebApplicationContext) : ControllerTest(context) {
    @Test
    @Throws(Exception::class)
    fun login() {
        mockMvc.perform(SecurityMockMvcRequestBuilders
                .formLogin("/login")
                .user("Admin")
                .password("@Admin0000")
        ).andExpect(MockMvcResultMatchers.status().`is`(HttpStatus.FOUND.value()))
            .andExpect(SecurityMockMvcResultMatchers.authenticated())
    }
}
